src/nacl/signing.py

```diff
@@ -97,17 +107,26 @@ def verify(self, smessage, signature=None, encoder=encoding.RawEncoder):
             signature and message concated together.
         :param signature: [:class:`bytes`] If an unsigned message is given for
             smessage then the detached signature must be provided.
-        :param encoder: A class that is able to decode the secret message and
-            signature.
+        :param encoder: (Deprecated) A class that is able to decode the secret
+            message and signature.
         :rtype: :class:`bytes`
         """
+        if encoder is None:
+            encoder = encoding.RawEncoder
+        else:
+            warn(("Automatic encoding/decoding of signed messages "
+                  "is deprecated and will be removed in a future release. "
+                  "Remove explicit 'encoder={}' calling parameter").format(
+                      encoder.__class__),
+                 PyNaclDeprecated)
+
         if signature is not None:
             # If we were given the message and signature separately, combine
             #   them.
-            smessage = signature + encoder.decode(smessage)
```

Comment:
Can we quickly pull this revert into its own PR so we can land that?

Reply:
I just reopened the Revert PR #505, and rebased it on current master.

Reply:
For some reason, #505 was kept in the Closed PR list. Just created #545 from the same branch.